src/BTHome.cpp

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2020 Particle Industries, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "BTHome.h"
+
+Vector<BTHome> BTHome::beacons;
+#define MAX_MANUFACTURER_DATA_LEN 37
+
+void BTHome::populateData(const BleScanResult *scanResult)
+{
+    Beacon::populateData(scanResult);
+    address = ADDRESS(scanResult);
+
+    uint8_t buf[BLE_MAX_ADV_DATA_LEN];
+    uint8_t count = ADVERTISING_DATA(scanResult).get(BleAdvertisingDataType::SERVICE_DATA, buf, BLE_MAX_ADV_DATA_LEN);
+
+    if (!parseBTHomeAdvertisement(buf, count))
+    {
+        Log.error("BTHome: advertisement parsing failed");
+    }
+}
+
+bool BTHome::isBeacon(const BleScanResult *scanResult)
+{
+    uint8_t buf[BLE_MAX_ADV_DATA_LEN];
+    uint8_t count = ADVERTISING_DATA(scanResult).get(BleAdvertisingDataType::SERVICE_DATA, buf, BLE_MAX_ADV_DATA_LEN);
+
+    if (count > 3 && buf[0] == 0xD2 && buf[1] == 0xFC) // BTHome UUID is 0xFCD2
```

Comment:
Please create an inline function to check for the UUID byte sequence belongs to this family.  I noticed the magic numbers here are also used elsewhere in this file.

Reply:
Hey Eric, can you elaborate a bit in the shape and form of that inline function, please? I'm not familiar. 
Is there an example elsewhere in the code of that?
Thanks

Reply:
For example:
```cpp
static inline bool BTHome::isBTHome(uint8_t lsb, uint8_t msb) const { return (0xD2 == lsb) && (0xFC == msb); } // BTHome UUID is 0xFCD2

if (count > 3 && isBTHome(buf[0], buf[1])) // BTHome UUID
```

Reply:
That would be reused on line 101

Reply:
got it, I addressed this in my next commit. Thanks